README.md

```diff
@@ -4,12 +4,34 @@ Engineering exercises around data: modeling, databases, storage of all kinds.
 
 
 ### Glossary of Terms
-#### Primary Key
+[This glossary](https://raima.com/database-terminology/) was a starting point for many of these definitions.
+
+#### Relational database
+A database in which inter-table relationships are organized primarily through common data columns, which define a one-to-many relationship between a row of the primary key table and one or more rows of the matching foreign key table. Joins relate tables that have matching primary/foreign key values, but other comparisons (relationships) may be defined. In addition to describing how the database tables are related, the relational model also defines how the related data can be accessed and manipulated. SQL is the most commonly used relational model database language.
+
 #### Table
+A collection of closely related columns. A table consists of rows each of which shares the same columns but vary in the column values.
+
 #### Column
-#### Relational database
-#### Index
+A single unit of named data that has a particular data type (e.g., number, text, or date). Columns only exist in tables.
+
 #### Data type
+The basic kind of data that can be stored in a column. The data types that are available in RDM SQL are: char, wchar, varchar, wvarchar, binary, varbinary, boolean, tinyint, smallint, integer, bigint, real, float, double, date, time, timestamp, long varbinary, long varchar, and long wvarchar. Postgres offers a [number of types](https://www.postgresql.org/docs/9.5/datatype.html).
```

Comment:
"RDM SQL" is not what we want -- that's a copy/paste that makes little sense here. RDM is the Raima Database Manager. Let's just make it "SQL".

If we're going to link to psql, probably also good to include https://www.sqlite.org/datatype3.html